url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/generic/IntegerExt.java

```diff
@@ -0,0 +1,88 @@
+package net.katsuster.ememu.generic;
+
+/**
+ * 整数演算のクラス
+ *
+ * @author katsuhiro
+ */
+public class IntegerExt {
+    /**
+     * キャリーが発生する（符号無し演算の加算がオーバーフローする）か、
+     * 否か、を取得します。
+     *
+     * @param left  被加算数
+     * @param right 加算する数
+     * @return キャリーが発生する場合は true、発生しない場合は false
+     */
+    public static boolean carryFrom(int left, int right) {
+        return compareUnsigned(left + right, left) < 0;
+    }
+
+    /**
+     * ボローが発生する（符号無し演算の減算がアンダーフローする）か、
+     * 否か、を取得します。
+     *
+     * @param left  被減算数
+     * @param right 減算する数
+     * @return キャリーが発生する場合は true、発生しない場合は false
+     */
+    public static boolean borrowFrom(int left, int right) {
+        return compareUnsigned(left, right) < 0;
+    }
+
+    /**
+     * オーバーフローが発生する（符号付き演算の結果が符号が変わる）か、
+     * 否か、を取得します。
+     *
+     * @param left  被演算数
+     * @param right 演算数
+     * @param add   加算なら true、減算なら false
+     * @return オーバーフローが発生したなら true、そうでなければ false
+     */
+    public static boolean overflowFrom(int left, int right, boolean add) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.